### PR TITLE
feat(ng-express-engine): support non-aot

### PR DESCRIPTION
**Pull request by FrozenPandaz**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/universal/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What modules are related to this pull-request**
- [x] Express Engine

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
remove unused proxy imports from some test files

Feature

* **What is the current behavior?** (You can also link to an open issue here)

Non-AoT Modules are not supported by the express engine

* **What is the new behavior (if this is a feature change)?**

Non-AoT Modules will be compiled, cached and then used for the request.

Module Factories which have been cached will be used for subsequent requests.

Bootstrapped Module Factories will be used to requests like normal 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

File: modules/ng-express-engine/src/file-loader.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as fs from 'fs';
+import { ResourceLoader } from '@angular/compiler';
+
+export class FileLoader implements ResourceLoader {
+  get(url: string): Promise<string> {
+    return new Promise((resolve) => {
+      resolve(fs.readFileSync(url).toString());
```
Comment (gdi2290) [inline modules/ng-express-engine/src/file-loader.ts]: if you're using a promise you might as well use the async API

File: modules/ng-express-engine/src/main.ts [hunk 2]
```diff
@@ -58,20 +74,52 @@ export function ngExpressEngine(setupOptions: NgSetupOptions) {
...
-      })
+      getFactory(moduleOrFactory, compiler)
+        .then(factory => {
+          return renderModuleFactory(factory, {
+            extraProviders: extraProviders
+          });
+        })
         .then((html: string) => {
           callback(null, html);
         });
```
Comment (gdi2290) [inline modules/ng-express-engine/src/main.ts]: you should also include an error callback to this promise

File: modules/ng-express-engine/src/main.ts [hunk 3]
```diff
@@ -58,20 +74,57 @@ export function ngExpressEngine(setupOptions: NgSetupOptions) {
...
+        .then(factory => {
+          return renderModuleFactory(factory, {
+            extraProviders: extraProviders
+          });
+        })
         .then((html: string) => {
           callback(null, html);
+        }, (err) => {
+          callback(err);
+          throw err;
```
Comment (gdi2290) [inline modules/ng-express-engine/src/main.ts]: you shouldn't have to throw here since the callback is handling it

Comment (FrozenPandaz) [inline modules/ng-express-engine/src/main.ts]: I was doing this to pass along the error, but i guess they could do it in the callback

**Pull request merged**